client/lib/post-normalizer/rule-content-make-images-safe.js

```diff
@@ -17,126 +12,83 @@ import { maxWidthPhotonishURL } from './utils';
-		return imageUrl && imageUrl.toLowerCase().indexOf( part ) === -1;
-	} );
-}
+	const safeSource = ( maxWidth
+		? maxWidthPhotonishURL( safeImageURL( imgSource ), maxWidth )
+		: safeImageURL( imgSource )
+		);
 
-export default function( maxWidth = false ) {
-	return function makeImagesSafe( post, dom ) {
-		let content_images = [],
-			images;
-
-		if ( ! dom ) {
-			throw new Error( 'this transform must be used as part of withContentDOM' );
-		}
-		images = dom.querySelectorAll( 'img[src]' );
-
-		// push everything, including tracking pixels, over to a safe URL
-		forEach( images, function( image ) {
-			let imgSource = image.getAttribute( 'src' ),
-				parsedImgSrc = url.parse( imgSource, false, true ),
-				hostName = parsedImgSrc.hostname;
-
-			let safeSource;
-			// if imgSource is relative, prepend post domain so it isn't relative to calypso
-			if ( ! hostName ) {
-				imgSource = url.resolve( post.URL, imgSource );
-				parsedImgSrc = url.parse( imgSource, false, true );
-			}
-
-			safeSource = safeImageURL( imgSource );
-			if ( ! safeSource && parsedImgSrc.search ) {
-				// we can't make externals with a querystring safe.
-				// try stripping it and retry
-				parsedImgSrc.search = null;
-				parsedImgSrc.query = null;
-				safeSource = safeImageURL( url.format( parsedImgSrc ) );
-			}
-
-			removeUnsafeAttributes( image );
-
-			if ( ! safeSource || imageShouldBeRemovedFromContent( imgSource ) ) {
-				image.parentNode.removeChild( image );
-				// fun fact: removing the node from the DOM will not prevent it from loading. You actually have to
-				// change out the src to change what loads. The following is a 1x1 transparent gif as a data URL
-				image.setAttribute( 'src', TRANSPARENT_GIF );
-				image.removeAttribute( 'srcset' );
-				return;
-			}
-
-			if ( maxWidth ) {
-				safeSource = maxWidthPhotonishURL( safeSource, maxWidth );
-			}
-
-			image.setAttribute( 'src', safeSource );
-
-			if ( image.hasAttribute( 'srcset' ) ) {
-				image.removeAttribute( 'srcset' );
-			}
-
-			if ( isCandidateForContentImage( imgSource ) ) {
-				content_images.push( {
-					src: safeSource,
-					original_src: imgSource,
-					width: image.width,
-					height: image.height
-				} );
-			}
-		} );
-
-		// grab all of the non-tracking pixels and push them into content_images
-		content_images = filter( content_images, function( image ) {
-			if ( ! image.src ) {
-				return false;
-			}
-			const edgeLength = image.height + image.width;
-			// if the image size isn't set (0) or is greater than 2, keep it
-			return edgeLength === 0 || edgeLength > 2;
-		} );
-
-		if ( content_images.length ) {
-			post.content_images = content_images;
-		}
-
-		return post;
-	};
+	removeUnwantedAttributes( image );
+
+	// trickery to remove it from the dom / not load the image
+	// TODO: test if this is necessary
+	if ( ! safeSource || imageShouldBeRemovedFromContent( imgSource ) ) {
+		image.remove();
```

Comment:
There's quite a few errors in the Calypso errors dashboard (path `/`) that I suspect trace back to this line:

```
Object doesn't support property or method 'remove'
```